crates/router/src/core/routing.rs

```diff
@@ -1111,3 +1148,173 @@ pub async fn update_default_routing_config_for_profile(
+                "unable to deserialize dynamic routing algorithm ref from business profile",
+            )?
+            .unwrap_or_default();
+
+        dynamic_routing_ref.update_algorithm_id(algorithm_id);
+        helpers::update_business_profile_active_dynamic_algorithm_ref(
+            db,
+            key_manager_state,
+            &key_store,
+            business_profile.clone(),
+            dynamic_routing_ref,
+        )
+        .await?;
+
+        let new_record = record.foreign_into();
+
+        metrics::ROUTING_CREATE_SUCCESS_RESPONSE.add(&metrics::CONTEXT, 1, &[]);
+        Ok(service_api::ApplicationResponse::Json(new_record))
+    } else {
+        let dynamic_routing_algo_ref: routing_types::RoutingAlgorithmRef = business_profile
+            .dynamic_routing_algorithm
+            .clone()
+            .map(|val| val.parse_value("DynamicRoutingAlgorithmRef"))
+            .transpose()
+            .change_context(errors::ApiErrorResponse::InternalServerError)
+            .attach_printable(
+                "unable to deserialize dynamic routing algorithm ref from business profile",
+            )?
+            .unwrap_or_default();
+
+        let timestamp = common_utils::date_time::now_unix_timestamp();
+
+        match dynamic_routing_algo_ref.algorithm_id {
+            Some(algorithm_id) => {
+                let dynamic_routing_algorithm = routing_types::DynamicRoutingAlgorithmRef {
+                    algorithm_id: None,
+                    timestamp,
+                };
+
+                let record = db
+                    .find_routing_algorithm_by_profile_id_algorithm_id(
+                        business_profile.get_id(),
+                        &algorithm_id,
+                    )
+                    .await
+                    .to_not_found_response(errors::ApiErrorResponse::ResourceIdNotFound)?;
+                let response = record.foreign_into();
+                helpers::update_business_profile_active_dynamic_algorithm_ref(
+                    db,
+                    key_manager_state,
+                    &key_store,
+                    business_profile,
+                    dynamic_routing_algorithm,
+                )
+                .await?;
+
+                metrics::ROUTING_UNLINK_CONFIG_SUCCESS_RESPONSE.add(&metrics::CONTEXT, 1, &[]);
+                Ok(service_api::ApplicationResponse::Json(response))
+            }
+            None => Err(errors::ApiErrorResponse::PreconditionFailed {
+                message: "Algorithm is already inactive".to_string(),
+            })?,
+        }
+    }
+}
+
+#[cfg(feature = "v1")]
+pub async fn dynamic_routing_update_configs(
+    state: SessionState,
+    request: routing_types::DynamicRoutingConfig,
+    algorithm_id: common_utils::id_type::RoutingId,
+    profile_id: common_utils::id_type::ProfileId,
+) -> RouterResponse<routing_types::RoutingDictionaryRecord> {
+    metrics::ROUTING_UPDATE_CONFIG_FOR_PROFILE.add(&metrics::CONTEXT, 1, &[]);
```

Comment:
It will be good to have metrics, even when it is an internal route (so that we can probably know which are the profiles for which we have updated configs).
But yeah added the profile_id as attribute.